python/ray/_private/usage/usage_lib.py

```diff
@@ -124,3 +154,81 @@ def get_cluster_metadata(gcs_client, num_retries) -> dict:
             num_retries=num_retries,
         )
     )
+
+
+def generate_report_data(cluster_metadata: dict) -> dict:
```

Comment:
I think we should use dataclasses or ideally pydantic data models for all of these, and only convert to JSON close to the wire (helps understand code & avoid compatibility tests)

Reply:
WE cannot use pydantic due to the minimal ray restriction. I think we can use dataclass here

Reply:
Used dataclasses